core/src/themes/ionic.mixins.scss

```diff
@@ -193,8 +193,22 @@
 
 @mixin rtl() {
   $root: #{&};
+  $selectors: str-split(#{add-root-selector($root, "[dir=rtl]")}, ",");
+  $dirSelector: "#{$root}:dir(rtl)";
+
+  // Selectors must be split into individual selectors in case the browser
+  // doesn't support a specific selector.
+  // For example, Firefox and Safari doesn't support :host-context().
+  // If an invalid selector is used, then the entire group of selectors
+  // will be ignored.
+  // @link https://www.w3.org/TR/selectors-3/#grouping
+  @each $selector in $selectors {
+    @at-root #{$selector} {
+      @content;
+    }
+  }
 
-  @at-root #{add-root-selector($root, "[dir=rtl]")} {
+  @at-root #{$dirSelector} {
```

Comment:
The browser support for [:dir](https://caniuse.com/css-dir-pseudo) is pretty limited. Did we decide to add this since Chrome has `:host-context`? Maybe we should add a comment on why this is here.

Reply:
It's more for a catch in case `:host-context()` doesn't work but the browser does support `:dir()`. I've added a comment as requested.